cpp/CMakeLists.txt

```diff
@@ -770,26 +771,31 @@ if(ARROW_WITH_BROTLI)
 endif()
 
 if(ARROW_WITH_BZ2)
+  list(APPEND ARROW_SHARED_PRIVATE_LINK_LIBS BZip2::BZip2)
```

Comment:
Why do you need shared libraries for third-party?
I don't think we use shared libraries for building tests, benchmarks, ORC inside the library.

Reply:
For tests/benchmarks, static-only is fine. But we do use shared libraries for other dependencies, and it would be nice to have that option. Admittedly, what I did here is not the best - so I'm happy to back out this commit in favor of a better solution.

Reply:
Can you describe the other dependencies within the library?

Reply:
Sorry if I wasn't clear - we would like the option to link libarrow et al. dynamically to its own dependencies, instead of being able to statically link only. It turns out this patch doesn't do anything anyways, so I'll revert it.